### ISSUE TITLE
Look into leveraging vscode.env.shell in remote workspaces

### ISSUE THREAD

**Issue by Tyriar** [Tyriar/vscode-windows-terminal#20]
Currently the SSH support works by **ignoring** which shell you selected and opening the host with the default shell. Perhaps instead we should pull `vscode.env.shell` which should be the default _remote_ shell and make that the default and potentially remove the other options?

FYI @ricardosantos9521

**Comment by ricardosantos9521** [comment]
Yes Windows terminal is using the default shell which can be a problem sometimes. Disabling selection would be the best scenario. We should try to use vscode.env.shell as the default windows terminal shell if we can else try to use the default. 

In the remote we should use vscode.env.shell if the remote is windows otherwise use we should use what we have already implemented.
Can we get the default shell of the remote workspace through vscode api when we are a ui extension?

**Comment by Tyriar** [comment]
> Can we get the default shell of the remote workspace through vscode api when we are a ui extension?

Yep, that's what `vscode.env.shell` will do. It is not possible to pull `/etc/shells` with SSH though without making our own connection, we could possibly do it if the SSH extension wanted to expose an API to do it. Maybe we should implement the remote shell detection in WSL first and if it works nicely explore options for SSH?

**Comment by ricardosantos9521** [comment]
vscode.env.shell returns always the terminal.integrated.shell.windows so i think we should use terminal.integrated.shell.windows and terminal.integrated.shell.linux as shells for
 wsl and ssh
